lgc/patch/VertexFetch.cpp

```diff
@@ -766,6 +773,11 @@ Function *VertexFetchImpl::generateFetchFunction(bool is64bitFetch, Module *modu
           shuffleMask.push_back(ConstantInt::get(Type::getInt32Ty(*m_context), i));
         vertex = builder.CreateShuffleVector(vertex, secondFetch, ConstantVector::get(shuffleMask));
       }
+
+      if (is16bitFetch) {
+        vertex = builder.CreateBitCast(vertex, FixedVectorType::get(Type::getInt16Ty(*m_context), 4));
+        vertex = builder.CreateZExt(vertex, fetch32Type);
+      }
```

Comment:
Please observe that `fetchVertex` has code to truncate the output of the fetch function in the 16-bit case. Instead of adding code to do a zero-extend here, it should be possible to just remove the truncation in the caller.

Also, the caller supports 8-bit fetches. I would expect those to use the 16-bit path here as well (and of course they would still be truncated).